src/server.cpp

```diff
@@ -1643,15 +1643,20 @@ void Server::SendInventory(PlayerSAO* playerSAO)
 void Server::SendChatMessage(u16 peer_id, const std::wstring &message)
 {
 	DSTACK(FUNCTION_NAME);
-
-	NetworkPacket pkt(TOCLIENT_CHAT_MESSAGE, 0, peer_id);
-	pkt << message;
-
 	if (peer_id != PEER_ID_INEXISTENT) {
+		std::wstring processed_message;
+		if (m_clients.getProtocolVersion(peer_id) < 27)
+			processed_message = unescape_enriched(message);
+		else
+			processed_message = message;
```

Comment:
i forget to tell you to remove this temporary variable and write directly to packet in the condition members